src/ConnectedMode.UnitTests/ServerSentEvents/ServerSentEventPumpTests.cs

```diff
@@ -0,0 +1,116 @@
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using SonarLint.VisualStudio.ConnectedMode.ServerSentEvents;
+using SonarLint.VisualStudio.Core.ServerSentEvents.Issues;
+using SonarLint.VisualStudio.Core.ServerSentEvents.TaintVulnerabilities;
+using SonarLint.VisualStudio.Integration.UnitTests;
+using SonarQube.Client.Models.ServerSentEvents;
+using SonarQube.Client.Models.ServerSentEvents.ClientContract;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.ServerSentEvents
+{
+    [TestClass]
+    public class ServerSentEventPumpTests
+    {
+        private Mock<IServerSentEventsSession> serverSentEventsSessionMock;
+        private Mock<IIssueChangedServerEventSourcePublisher> issueChangedPublisherMock;
+        private Mock<ITaintServerEventSourcePublisher> taintPublisherMock;
+        private ServerSentEventPump testSubject;
+
+        [TestInitialize]
+        public void SetUp()
+        {
+            serverSentEventsSessionMock = new Mock<IServerSentEventsSession>();
+            issueChangedPublisherMock = new Mock<IIssueChangedServerEventSourcePublisher>();
+            taintPublisherMock = new Mock<ITaintServerEventSourcePublisher>();
+
+            testSubject = new ServerSentEventPump(issueChangedPublisherMock.Object, taintPublisherMock.Object);
+        }
+
+        [TestMethod]
+        public void Dispose_DisposesPublishers()
+        {
+            testSubject.Dispose();
+
+            issueChangedPublisherMock.Verify(ch => ch.Dispose(), Times.Once);
+            taintPublisherMock.Verify(ch => ch.Dispose(), Times.Once);
+        }
+
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<ServerSentEventPump, IServerSentEventPump>(
+                MefTestHelpers.CreateExport<IIssueChangedServerEventSourcePublisher>(),
+                MefTestHelpers.CreateExport<ITaintServerEventSourcePublisher>());
+        }
+
+        [TestMethod]
+        public async Task PumpAllAsync_SelectsPublisherCorrectlyAndPreservesOrderWithinType()
+        {
+            var inputSequence = new IServerEvent[]
+            {
+                Mock.Of<IIssueChangedServerEvent>(),
+                Mock.Of<ITaintVulnerabilityRaisedServerEvent>(),
+                Mock.Of<IIssueChangedServerEvent>(),
+                Mock.Of<ITaintVulnerabilityClosedServerEvent>(),
+                Mock.Of<ITaintVulnerabilityClosedServerEvent>(),
+                Mock.Of<IIssueChangedServerEvent>(),
+                Mock.Of<ITaintVulnerabilityRaisedServerEvent>(),
+            };
+            SetUpEventsSequenceThatFinishes(inputSequence);
+
+            await testSubject.PumpAllAsync(serverSentEventsSessionMock.Object);
+            
+            issueChangedPublisherMock.Invocations
```

Comment:
I think it would be more readable to specify exactly what you're expecting.
It's better to be explicit in the tests.
```
issueChangedPublisherMock.Invocations.Count.Should().Be(3)
issueChangedPublisherMock.Invocations[0].Arguments[0].Should().Be(inputSequence[0])
```